File: frontend/src/features/trails/types/trail.ts
export type TrailDifficulty = "Easy" | "Intermediate" | "Advanced" | "Expert";

export type RawTrailDifficulty = TrailDifficulty | number;

export interface TrailResponse {
  id: string;
  docId: string;
  name: string;
  city: string;
  region: string;
  difficulty: TrailDifficulty;
  distanceKm: number;
  description: string;
  imageUrl?: string | null;
  latitude?: number | null;
  longitude?: number | null;
  createdAt: string;
  updatedAt: string;
}

export interface RawTrailResponse {
  id: string;
  docId: string;
  name: string;
  city: string;
  region: string;
  difficulty: RawTrailDifficulty;
  distanceKm: number;
  description: string;
  imageUrl?: string | null;
  latitude?: number | null;
  longitude?: number | null;
  createdAt: string;
  updatedAt: string;
}

export interface TrailQueryRequest {
  search?: string;
  difficulty?: TrailDifficulty;
  pageNumber?: number;
  pageSize?: number;
}

export interface PagedTrailResponse {
  items: TrailResponse[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}

export interface RawPagedTrailResponse {
  items: RawTrailResponse[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}

File: backend/Entities/Trail.cs
using backend.Enums;

namespace backend.Entities;

public class Trail
{
    public Guid Id { get; set; }

    public string DocId { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string City { get; set; } = string.Empty;

    public string Region { get; set; } = string.Empty;

    public TrailDifficulty Difficulty { get; set; }

    public decimal DistanceKm { get; set; }

    public string Description { get; set; } = string.Empty;

    public double? CoordinateX { get; set; }

    public double? CoordinateY { get; set; }

    public double? Latitude { get; set; }

    public double? Longitude { get; set; }

    public bool IsActive { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public ICollection<CheckIn> CheckIns { get; set; } = [];
}
